File: TodayFitComplete_Backend/src/main/java/com/SJY/TodayFitComplete_Backend/dto/file/FileUploadResponse.java
package com.SJY.TodayFitComplete_Backend.dto.file;

import com.SJY.TodayFitComplete_Backend.entity.file.FileEntity;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 파일 업로드 후 응답 DTO.
 */
@Getter
@Setter
@NoArgsConstructor
public class FileUploadResponse {

    private Long fileId;
    private String originFileName;
    private String filePath;
    private String fileType;

    @Builder
    public FileUploadResponse(Long fileId, String originFileName, String filePath, String fileType) {
        this.fileId = fileId;
        this.originFileName = originFileName;
        this.filePath = filePath;
        this.fileType = fileType;
    }

    public static FileUploadResponse fromEntity(FileEntity file) {
        return FileUploadResponse.builder()
                .fileId(file.getId())
                .originFileName(file.getOriginFileName())
                .filePath(file.getFilePath())
                .fileType(file.getFileType())
                .build();
    }
}

File: TodayFitComplete_Frontend/app/src/main/java/kr/ac/kumoh/ce/s20190633/todayfitcomplete_frontend/Dto/File/FileUploadResponse.kt
package kr.ac.kumoh.ce.s20190633.todayfitcomplete_frontend.Dto.File

// FileUploadResponse 클래스: 파일 업로드 후 서버에서 반환되는 응답 데이터를 나타냅니다.
data class FileUploadResponse(
    val fileId: Long,             // 파일의 고유 ID
    val originFileName: String,   // 업로드된 파일의 원본 파일 이름
    val filePath: String,         // 업로드된 파일의 서버 상의 저장 경로
    val fileType: String          // 파일의 유형 (예: 이미지, 문서 등)
)
